sdk/extensions/Azure.Extensions.AspNetCore.Configuration.Secrets/src/AzureKeyVaultConfigurationProvider.cs

```diff
@@ -127,13 +127,25 @@ await foreach (var secret in secretPages.ConfigureAwait(false))
 
         private void SetData(Dictionary<string, LoadedSecret> loadedSecrets, bool fireToken)
         {
-            var data = new Dictionary<string, string>(loadedSecrets.Count, StringComparer.OrdinalIgnoreCase);
+            var data = new Dictionary<string, LoadedSecret>(loadedSecrets.Count, StringComparer.OrdinalIgnoreCase);
             foreach (var secretItem in loadedSecrets)
             {
-                data.Add(secretItem.Value.Key, secretItem.Value.Value);
+                string key = secretItem.Value.Key;
+
+                if (data.TryGetValue(key, out LoadedSecret currentSecret))
```

Comment:
Can you, please add a comment here about what situation we are trying to handle.

Reply:
Good point @pakrym  :) A comment has been added.